### PR TITLE
Improve admin UX

### LINKED ISSUE #12: Add link in admin to create campaign in goteo
image

### LINKED ISSUE #13: API User and Key configuration per component
As an admin, I can

- [x] Configure goteo API user and password for each social crowdfunding campaign component
- [x] Configure how often the campaign will be automatically updated from Goteo API

### PR DESCRIPTION
**Pull request by verarojman**
image

Closes #12 
Closes #13 

- [x] Add specs

### CODE REVIEW

File: app/models/decidim/social_crowdfunding/campaign.rb [hunk 1]
```diff
@@ -40,17 +40,30 @@ def self.params_from_json(json)
...
+          else
+            campaign = create!(params_from_json(json).merge(slug: slug, organization: component.organization))
+          end
         end
 
         campaign
       end
 
+      def self.should_sync?(campaign, component)
+        campaign.created_at < 1.minute.ago || campaign.updated_at > component.settings.goteo_api_update_hours.hours.ago
```
Comment (microstudi) [inline app/models/decidim/social_crowdfunding/campaign.rb:64]: Why `< 1.minute.ago` ?

File: lib/decidim/social_crowdfunding/goteo/api.rb [hunk 2]
```diff
@@ -5,34 +5,39 @@ module SocialCrowdfunding
     module Goteo
       module Api
         PROJECT_URL = "%{base_url}/projects/%{slug}"
-        PROJECTS_URL = "%{base_url}/projects/?limit=%{limit}&page=%{page}"
 
         class << self
           def base_url
-            "https://api.goteo.org/v1"
+            ENV.fetch("GOTEO_BASE_URL", "https://api.goteo.org/v1")
```
Comment (microstudi) [inline lib/decidim/social_crowdfunding/goteo/api.rb:11]: That's ok, but i'd use a configuration option for the plugin in the future. Maybe add a Fixme comment

File: app/controllers/decidim/social_crowdfunding/admin/campaigns_controller.rb [hunk 3]
```diff
@@ -5,10 +5,31 @@ module SocialCrowdfunding
...
       class CampaignsController < Decidim::Admin::Components::BaseController
         include Decidim::Paginable
+        include Decidim::SocialCrowdfunding::HasCampaign
+
+        helper Decidim::SocialCrowdfunding::Admin::ApplicationHelper
 
         helper_method :campaigns
 
-        def index; end
+        def index
```
Comment (microstudi) [inline app/controllers/decidim/social_crowdfunding/admin/campaigns_controller.rb:14]: maybe check some permissions here?

**Review by microstudi — APPROVED** [review]
Just a couple of notes but this looks amazing!

File: app/models/decidim/social_crowdfunding/campaign.rb (hunk 1 above)
Comment (verarojman) [reply app/models/decidim/social_crowdfunding/campaign.rb:64]: Good question, I'll look into it tomorrow

File: app/controllers/decidim/social_crowdfunding/admin/campaigns_controller.rb (hunk 3 above)
Comment (verarojman) [reply app/controllers/decidim/social_crowdfunding/admin/campaigns_controller.rb:14]: Sure, I had planned to implement permissions it in a different PR (see #21)

File: lib/decidim/social_crowdfunding/goteo/api.rb (hunk 2 above)
Comment (verarojman) [reply lib/decidim/social_crowdfunding/goteo/api.rb:11]: Can you share the link with an example? I couln't find a reference for the way we usually set this up.

Comment (microstudi) [reply lib/decidim/social_crowdfunding/goteo/api.rb:11]: Also a great way to have options for different configuration and explain them. https://github.com/Platoniq/decidim-module-civicrm/blob/main/lib/decidim/civicrm.rb

**Pull request merged**